src/routes/Profile/Profile.tsx

```diff
@@ -102,6 +117,7 @@ const Profile: React.FunctionComponent = () => {
 
   useEffect(() => {
     setPageTitle('Profile', wallet.account);
+    invalidateCache();
```

Comment:
why do we invalidate cache in here?

Reply:
it is a component did mount again. Maybe they went to other pages and now return to the profile page to check for badge progress

Reply:
oh i see, this essentially accounts for the scenario where they e.g. have done a trade on the trading pools page and then returned to the profile page they also see their most recent badges

Reply:
what if instead of invalidating the entire badge cache, we edit to the list of badges?

Reply:
also, note in order for a badge to be registered, the on-chain event data needs to be indexed by the subgraph which then feeds into the FE, however the indexing of new events takes some time and i suspect there are edgecases where users have to wait (or even refresh) the profile in order see the most recent set of their badges that accounts for their latest interactions with the protocol

Reply:
```what if instead of invalidating the entire badge cache, we edit to the list of badges?``` We still have to fetch the array, the point behind cache is that we don't do API calls

Reply:
yeah, that's along the lines of what i'm thinking as well

Reply:
need to edit the list, it can either increase in size or occasionally decrease in size in case we realise there is an issue with the way subgraph computes badge eligibility (thought this should not happen in theory)